ui/src/components/Feed/FeedPost.svelte

```diff
@@ -58,30 +56,35 @@
       return api.portal.do.subscribe(key);
     }
 
-    // recursively fetch replies so we can display the total number of replies per OP
-    const getNestedReplies = (replies) => {
-      if (Array.isArray(replies) && replies.length > 0) {
-        for (let reply of replies) {
-          if (!allReplies.has(reply)) {
-            allReplies.add(reply);
-            const repliesToReply = getReplies(reply);
-            if (repliesToReply) {
-              getNestedReplies(repliesToReply);
-            }
-          }
-        }
-      }
-    };
-
-    getNestedReplies(getReplies(key));
-
     let likes = [...(getLikes(key.ship, key) || [])];
 
     numLikes = likes.length;
     if (isLikedByMe && !likes.find((l) => l.ship === me)) numLikes++;
     isLikedByMe = isLikedByMe || !!likes.find((l) => l.ship === me);
   };
 
+  // recursively fetch replies so we can display the total number of replies per OP
+  const getNestedReplies = async (replies) => {
+    if (Array.isArray(replies) && replies.length > 0) {
+      for (let reply of replies) {
+        if (!allReplies.has(reply)) {
+          allReplies.add(reply);
+          const repliesToReply = getIndexerAndLocalReplies(reply);
+          if (repliesToReply) {
+            await getNestedReplies(repliesToReply);
+          }
+        }
+      }
+    }
+    allReplies = allReplies;
+  };
+
+  const loadNestedReplies = async () => {
+    await getNestedReplies(getIndexerAndLocalReplies(key));
+  };
+
+  $: $state && loadNestedReplies();
```

Comment:
the `$:` statement in general basically says "if any variable inside this statement changes, re-run the statement", which is why we need to do `$: $state && ...` here, to get the statement to re-run when the state changes, it's the same as doing `state.subscribe(loadNestedReplies)` but imo a little nicer.

Reply:
I see, thanks for fixing this. so this needed the reassignment and async function to work. it was already reactive in the last PR since getNestedReplies was called inside loadPost. that design was a bit cleaner

Reply:
none of this is async, i'm not sure where that came from